File: 3646.java
//* @author:
import java.util.*;
public   class Main{

    public static void main(String[] args) {
       Scanner sc=new Scanner(System.in);

    int i,j,ans;
    end: while (sc.hasNext()) {
        int n=sc.nextInt();
        if(n==0) break;
        int m=sc.nextInt();
        int a[]=new int[n+1];
        int b[]=new int[m+1];

        for ( i=1;i<=n;i++)
           a[i]=sc.nextInt();
        for ( i=1;i<=m;i++)
           b[i]=sc.nextInt();
        Arrays.sort(a, 1,n+1);
        Arrays.sort(b,1,m+1);
        for (ans=0,j=i=1;i<=n;i++) {
        if (j>m) {
             System.out.println("Loowater is doomed!");
              continue end;
         }
            while (a[i]>b[j]) {
              j++;if (j>m) {
                System.out.println("Loowater is doomed!");
                continue end;
              }
            }
            ans+=b[j];j++;
        }
        System.out.printf("%d\n",ans);

    }
  }
}

File: 3646.cc
#include<cstdio>
#include<cstring>
#include<iostream>
#include<algorithm>
#define MAX 20001
using namespace std;
int n,m,a[MAX],b[MAX];
int main()
{
    while(scanf("%d%d",&n,&m)&&(n||m))
    {
	for(int i=0;i<n;i++)
	    scanf("%d",&a[i]);
	for(int i=0;i<m;i++)
	    scanf("%d",&b[i]);
	sort(a,a+n);
	sort(b,b+m);
	int ans=0,cou=0,i;
	for(i=0;i<n;i++)
	{
	    while(cou<m&&b[cou]<a[i])
		cou++;
	    if(cou<m)
		ans+=b[cou++];
	    else
		break;
	}
	if(i==n)
	    printf("%d\n",ans);
	else
	    printf("Loowater is doomed!\n");
    }
    return 0;
}


